esp/platform/txsummary.hpp

```diff
@@ -50,80 +75,264 @@ class CTxSummary : extends CInterface
+            : TxEntryNumber(_key, _logLevel, _group, _suffix), value(_value) {}
+        virtual TxEntryBase* cloneWithNewName(const char* newName = nullptr) override {return new TxEntryUInt(newName ? newName : fullname.get(), value, logLevel, group, suffix);}
+        virtual StringBuffer& appendValue(StringBuffer& buf) override {return buf.append(value);}
+    };
+
+    struct TxEntryUInt64 : public TxEntryNumber
+    {
+        unsigned __int64 value;
+
+        TxEntryUInt64(const char* _key, unsigned __int64 _value, const LogLevel _logLevel, const unsigned int _group = DEFAULT_GROUP, const char* _suffix = nullptr)
+            : TxEntryNumber(_key, _logLevel, _group, _suffix), value(_value) {}
+        virtual TxEntryBase* cloneWithNewName(const char* newName = nullptr) override {return new TxEntryUInt64(newName ? newName : fullname.get(), value, logLevel, group, suffix);}
+        virtual StringBuffer& appendValue(StringBuffer& buf) override {return buf.append(value);}
+    };
+
+    struct TxEntryDouble : public TxEntryNumber
+    {
+        double value;
+
+        TxEntryDouble(const char* _key, double _value, const LogLevel _logLevel, const unsigned int _group = DEFAULT_GROUP, const char* _suffix = nullptr)
+            : TxEntryNumber(_key, _logLevel, _group, _suffix), value(_value) {}
+        virtual TxEntryBase* cloneWithNewName(const char* newName = nullptr) override {return new TxEntryDouble(newName ? newName : fullname.get(), value, logLevel, group, suffix);}
+        virtual StringBuffer& appendValue(StringBuffer& buf) override {return buf.append(value);}
+    };
+
+    struct TxEntryTimer : public TxEntryBase
+    {
+        // The CumulativeTimer object owned by this Entry needs to manage
+        // the unsigned int and LogLevel values. This is because external
+        // users of the TxSummary class expect to be able to set those values
+        // on the CumulativeTimer object. So the logLevel and group values
+        // in the TxEntryBase are not used for this subclass.
+
+        Owned<CumulativeTimer> value;
+
+        TxEntryTimer(const char* _key, CumulativeTimer* _value, const LogLevel _logLevel = LogMin, const unsigned int _group = DEFAULT_GROUP)
+            : TxEntryBase(_key, _logLevel, _group)
+        {
+            value.set(_value);
+        }
+        virtual TxEntryBase* cloneWithNewName(const char* newName = nullptr) override {return new TxEntryTimer(newName ? newName : fullname.get(), value.get(), logLevel, group);}
+
+
+        // Uses logLevel and group settings stored in the contained CumulativeTimer object
+        virtual bool shouldSerialize(const LogLevel requestedLevel, const unsigned int requestedGroup) override;
+        virtual StringBuffer& serialize(StringBuffer& buf, const LogLevel requestedLevel, const unsigned int requestedGroup, const unsigned int requestedStyle) override;
+    };
+
+    struct TxEntryObject;
```

Comment:
Why have a forward declaration when it is not used between here and the actual declaration?